drivers/disk/usdhc.c

```diff
@@ -2596,6 +2611,34 @@ static int usdhc_sd_init(struct usdhc_priv *priv)
 
 static K_MUTEX_DEFINE(z_usdhc_init_lock);
 
+
+static int usdhc_dat3_pull(bool pullup, struct usdhc_priv *priv)
+{
+	int ret = 0U;
+
+	if (pullup) {
+		/* Just call the board specific function (registered in SOC file) */
+		imxrt_usdhc_dat3_pull(pullup);
+	} else {
+		/* Call board specific function to pull down DAT3 */
+		imxrt_usdhc_dat3_pull(pullup);
+		/* Power off the card to clear DAT3 legacy status */
+		if (priv->pwr_gpio) {
+			ret = gpio_pin_set(priv->pwr_gpio, priv->config->pwr_pin, 0);
+			if (ret) {
+				return ret;
+			}
+			/* Delay for card power off to complete */
+			usdhc_millsec_delay(1000);
```

Comment:
a bit of a magic number here. Should this be an internal Kconfig value?

Reply:
This is pulled from the board specific SDMMC file. I changed the delay to be a Kconfig value, and the RT1170 EVK defconfig sets it to 1000.